perceiver/model/image/optical_flow.py

```diff
@@ -0,0 +1,227 @@
+import os
+from dataclasses import dataclass
+from typing import Tuple
+
+import numpy as np
+import torch
+import torch.nn as nn
+from transformers import PerceiverConfig as HuggingfacePerceiverConfig, PerceiverForOpticalFlow
+from transformers.models.perceiver.modeling_perceiver import PerceiverImagePreprocessor, space_to_depth
+
+from perceiver.model.core import (
+    EncoderConfig,
+    InputAdapter,
+    OutputAdapter,
+    PerceiverConfig,
+    PerceiverDecoder,
+    PerceiverEncoder,
+    PerceiverIO,
+)
+from perceiver.model.core.config import DecoderConfig
+from perceiver.model.core.convert import (
+    copy_cross_attention_layer_params,
+    copy_param,
+    copy_params,
+    copy_self_attention_block_params,
+)
+from perceiver.model.image.common import FourierPositionEncoding
+
+
+@dataclass
+class OpticalFlowEncoderConfig(EncoderConfig):
+    image_shape: Tuple[int, int] = (368, 496)
+    num_temporal_channels: int = 2
+    num_image_input_channels: int = 27
+    num_image_output_channels: int = 64
+    num_frequency_bands: int = 64
+    spatial_downsample: int = 1
+    temporal_downsample: int = 2
+
+
+@dataclass
+class OpticalFlowDecoderConfig(DecoderConfig):
+    output_image_shape: Tuple[int, int] = (368, 496)
+    num_output_image_channels: int = 2
```

Comment:
Is this ever other than `2`?

Reply:
No, will be converted to a constructor variable of `OpticalFlowOutputAdapter`